### DISCUSSION TITLE
Unsign models

### DISCUSSION THREAD

**Discussion by juliangarcess** [creagia/laravel-sign-pad#14] [Ideas]
I think it would be great if we have a method on the package for unsign a model.

**Comment by dtorras** [comment]
Good idea. If you want to implement it, I think I prefer to create a new `deleteSignature()` in the `RequiresSignature` concern https://github.com/creagia/laravel-sign-pad/blob/main/src/Concerns/RequiresSignature.php

**Reply by juliangarcess** [reply]
I think we can use the getSignatureRoute and make a request using HTTP delete and add a method to the controller for that case

**Reply by juliangarcess** [reply]
what do you think about that? @dtorras

**Reply by dtorras** [reply]
I prefer not to create new routes and leave that on the userland. With a new `deleteSignature()` method, user can call the `$model->deleteSignature()` method from any part of their application, with or without a dedicated route, controller or request.

**Reply by juliangarcess** [reply]
but that is the same of using $model->signature?->delete()

**Reply by dtorras** [reply]
Yes, exactly. Just a shortcut for convenience.

**Reply by juliangarcess** [reply]
okay 👍 I push a PR later

**Reply by juliangarcess** [reply]
I have pushed the PR
